requirements-gpu.txt

```diff
@@ -3,4 +3,4 @@
 # cudf>=21.12
 # dask-cudf>=21.12
 # dask-cuda>=21.12
-# cupy>=7
+cupy-cuda11x
```

Comment:
I think the version does matter here, since CuPy didn't add support for bools with DLpack until 12.0.0

Reply:
We cant, when you try to specify that here it fails... When I try something like `pip install cupy-cuda11x>=12.0.0 it silently fails and nothing gets installed. This is after I have previously uninstalled the version in the environment. This is why I just left it to grab the latest version. 
<img width="309" alt="image" src="https://user-images.githubusercontent.com/37191411/230633483-76a11757-6e7c-45dc-910f-97cc8a79b545.png">

Reply:
So you end up being forced to run the plain `pip install cupy-cuda11x` and you get the latest version.

Reply:
I don't think we want to install `cupy-cuda11x` specifically though. In the containers we do, but here the requirement is actually to have `cupy>=12.0.0`—which we will by installing `cupy-cuda11x`, but would also be satisfied by `cupy-cuda12x`, `cupy-cuda102`, etc.

Reply:
Okay, I'm wrong about that. `pip install cupy>=12.0.0` still tries to install from source even if you have `cupy-cuda*>=12.0.0`. Maybe we could make the various packages for satisfying the cupy dependency optional instead?